jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -108,17 +113,54 @@ public boolean release()
     }
 
     /**
-     * @return whether this instance is retained
-     * @see ReferenceCounter#isRetained()
+     * <p>Returns a {@code RetainableByteBuffer} that wraps
+     * the given {@code ByteBuffer} and {@link Runnable} releaser.</p>
+     *
+     * @param byteBuffer the {@code ByteBuffer} to wrap
+     * @param releaser a {@link Runnable} to call when the buffer is released.
+     * @return a {@code RetainableByteBuffer}
      */
-    boolean isRetained();
+    static RetainableByteBuffer wrap(ByteBuffer byteBuffer, Runnable releaser)
+    {
+        return new AbstractRetainableByteBuffer(byteBuffer)
+        {
+            {
+                acquire();
+            }
+
+            @Override
+            public boolean release()
+            {
+                boolean released = super.release();
+                if (released)
+                    releaser.run();
+                return released;
+            }
+        };
+    }
 
     /**
      * Get the wrapped, not {@code null}, {@code ByteBuffer}.
      * @return the wrapped, not {@code null}, {@code ByteBuffer}
      */
     ByteBuffer getByteBuffer();
 
+    /**
+     * Creates a copy of this RetainableByteBuffer that is entirely independent, but
+     * backed by the same memory space, i.e.: modifying the ByteBuffer of the original
+     * also modifies the ByteBuffer of the copy and vice-versa.
+     * @return A copy of this RetainableByteBuffer
+     */
+    default RetainableByteBuffer copy()
+    {
+        return new AbstractRetainableByteBuffer(BufferUtil.copy(getByteBuffer()))
```

Comment:
Either the comment or the implementation is wrong.   Comment implies shallow copy, but implementation is a deep copy!